src/plugin.rs

```diff
@@ -1,14 +1,54 @@
+use crate::PlaySoundEvent;
 use bevy::app::{App, Plugin};
+use bevy::audio::AudioSinkPlayback;
 use bevy::log::{debug, trace};
-use bevy::prelude::{EventReader, NonSend, PostUpdate, Resource, Startup, Update, World};
+use bevy::math::Vec3;
+use bevy::prelude::{
+    Added, Commands, Component, Entity, EventReader, GlobalTransform, NonSend, PostUpdate, Query,
+    Res, Resource, Startup, Update, World,
+};
+use bevy::time::Time;
 use bevy_mod_sysfail::sysfail;
 use libfmod::ffi::{FMOD_INIT_NORMAL, FMOD_STUDIO_INIT_NORMAL, FMOD_STUDIO_LOAD_BANK_NORMAL};
-use libfmod::{EventDescription, Studio};
-use play_sound_event::PlaySoundEvent;
+use libfmod::{Attributes3d, EventDescription, Studio, Vector};
 use std::env::var;
 use std::fs::{canonicalize, read_dir};
 use std::path::{Path, PathBuf};
 
+#[derive(Component)]
+pub struct FmodListener {
+    previous_position: Vec3,
+}
+
+impl Default for FmodListener {
+    fn default() -> Self {
+        FmodListener {
+            previous_position: Vec3::default(),
+        }
+    }
+}
+
+#[derive(Component)]
+pub struct FmodAudioSource {
+    pub name: &'static str,
+}
+
+// This should be OK I think since we should only have one event instance per entity
+unsafe impl Sync for UnsafeEventInstance {}
+unsafe impl Send for UnsafeEventInstance {}
+
+// todo: should not be pub eventually
```

Comment:
`UnsafeEventInstance` should not be pub imo because it's something internal to the plugin that the user shouldn't have to interact with if we make the API nice enough. Some time after I placed this comment I implement Drop so it doesn't need to be pub anymore, just forgot to address the comment